### PR TITLE
[SPARK-40219][SQL] Resolved view logical plan should hold the schema to avoid redundant lookup

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

In `CatalogImpl`, we need to look up the view again to get the schema, as `ResolvedView` only contains the view identifier. This PR fix this issue by refactoring the `ResolvedView`:
1. Split it into `ResolvedPersistentView` and `ResolvedTempView`
2. The new logical plans hold the view schema
3. `ResolvedPersistentView` holds the catalog, to match `ResolvedTable`.

This PR also cleans up the temp view related methods in `SessionCatalog`:
1. group them together in the section `Methods that interact with temp views only`
2. rename 2 methods to make the naming more consistent.

### Why are the changes needed?

code cleanup and avoid redundant view lookup

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

existing tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @viirya @huaxingao @MaxGekk

**Comment by viirya** [comment]
Seems many test failures.

**Comment by cloud-fan** [comment]
@viirya tests all passed now

**Comment by viirya** [comment]
Thanks @cloud-fan. I'll review this today.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 1]
```diff
@@ -1096,46 +1099,44 @@ class Analyzer(override val catalogManager: CatalogManager)
...
+          val target = if (tempViewPlan.isStreaming) "streams" else "views"
           throw QueryCompilationErrors.timeTravelUnsupportedError(target)
         }
+        tempViewPlan
       }
-      tmpView
     }
 
     /**
      * Resolves relations to `ResolvedTable` or `ResolvedView`. This is for resolving DDL and
```
Comment (viirya) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala:1134]: ResolvedView -> ResolvedTempView and ResolvedPersistentView

**Review by viirya — APPROVED** [review]

**Review by huaxingao — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks for review, merging to master!

**Pull request closed without merging**